# DESeqPLUS

### Description:
`DESeqPLUS` provides a set of functions to simplify the visualization of RNA-seq differential expression results generated by DESeq2. The package includes tools for quality control summaries, PCA exploration, and volcano plots. By combining these functions into a single package, it streamlines the workflow for examining DESeq2 outputs, enabling users to quickly identify patterns, check data quality, and interpret results without manually coding each plot. Unlike other tools that focus solely on statistical analysis, `DESeqPLUS` emphasizes visual interpretation and results, improving the accessibility and efficiency of RNA-seq data analysis. This package was developed using R version 4.5.1 on macOS Sequoia 15.2.

### Installation:
To install the latest version of the package:
```
install.packages("devtools")
library("devtools")
devtools::install_github("katfishy/DESeqPLUS", build_vignettes = TRUE)
library("DESeqPLUS")
```

To run the shinyApp:
```
runDESeqPLUS()
```

### Overview:
```
ls("package:DESeqPLUS")
data(package = "DESeqPLUS") 
browseVignettes("DESeqPLUS")
```
`DESeqPLUS` currently contains 3 functions:
1. **qcPlot** for producing quality control plots that include total library size per sample and the distribution of log-transformed counts, to help assess sample quality before running differential gene analysis.
2. **pcaPlot** for conducting principal component analysis on normalized counts and visualize sample clustering, alllowing users to identify patterns before running differential gene analysis.
3. **volcanoPlot** for producing a volcano plot to visualize differentially expressed genes, plotting log2 fold changes against -log10 adjusted p-values to highlight significant genes.

The package also contains the **airway** dataset with read counts of four human airway smooth muscle cell lines treated with dexamethasone. An overview of the package is illustrated below.

<img width="578" height="403" alt="Screenshot 2025-11-04 at 8 42 39 PM" src="https://github.com/user-attachments/assets/9f0aacc5-9f96-4b57-a4c8-7190613b8f67" />

### Contributions:
The author of the package is Katie Lee. The author wrote all functions currently included in the package: `qcPlot()`, `pcaPlot()`, and `volcanoPlot()`.
- `qcPlot()` generates  quality control plots, including total library size per sample and log-transformed count distributions. This function makes use of the `counts()` function from the **DESeq2** package and plotting functions from **ggplot2**. Base R functions are used for data reshaping and calculation of library sizes. 

- `pcaPlot()` performs principal component analysis on normalized counts using `varianceStabilizingTransformation()` from **DESeq2**, `assay()` and `colData()` from **SummarizedExperiment**, `prcomp()` from base R, and **ggplot2** for visualization, allowing exploration of sample clustering and outliers.

- `volcanoPlot()` creates a volcano plot to visualize differentially expressed genes using log2 fold changes and -log10 adjusted p-values. This function relies on **ggplot2** for plotting and uses base R for data manipulation.

The **airway** dataset was used throughout the package for examples, testing, and the vignette. Generative AI tools (ChatGPT) were used to assist with brainstorming, debugging, and optimization of documentations. All code suggestions from AI were reviewed by the author.

### References:
- BioRender. (2025). Image created by Lee, K. Retrieved November 4, 2025, from https://app.biorender.com/
- Himes, E. B, Jiang, X., Wagner, P., Hu, R., Wang, Q., Klanderman, B., Whitaker, M. R, Duan, Q., Lasky-Su, J., Nikolos, C., Jester, W., Johnson, M., Panettieri, A. R, Tantisira, G. K, Weiss, T. S, Lu, Q. (2014). “RNA-Seq Transcriptome Profiling Identifies CRISPLD2 as a Glucocorticoid Responsive Gene that Modulates Cytokine Function in Airway Smooth Muscle Cells.” PLoS ONE, 9(6), e99625. http://www.ncbi.nlm.nih.gov/pubmed/24926665.
- Love, M.I., Huber, W. & Anders, S. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biol 15, 550 (2014). https://doi.org/10.1186/s13059-014-0550-8
- Morgan M, Obenchain V, Hester J, Pagès H (2025). SummarizedExperiment: A container (S4 class) for matrix-like assays. doi:10.18129/B9.bioc.SummarizedExperiment, R package version 1.40.0, https://bioconductor.org/packages/SummarizedExperiment.
- OpenAI. (2025). ChatGPT (GPT-5) [Large language model] https://chat.openai.com/
- Wickham H (2016). ggplot2: Elegant Graphics for Data Analysis. Springer-Verlag New York. ISBN 978-3-319-24277-4, https://ggplot2.tidyverse.org.

### Acknowledgements:
This package was developed as part of an assessment for 2025 BCB410H: Applied Bioinformatics course at the University of Toronto, Toronto, CANADA. TestingPackage welcomes issues, enhancement requests, and other contributions. To submit an issue, use the GitHub issues. Many thanks to those who provided feedback to improve this package.
